src/sonic-config-engine/minigraph.py

```diff
@@ -1033,6 +1033,11 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         if port:
             port['admin_status'] = 'up'
 
+    for port in ports.keys():
```

Comment:
Just to confirm, `ports `is only from the `DeviceInterfaceLinks `and not the `EthernetInterfaces `from `DeviceInfo`

Reply:
`ports` is coming from the function `get_port_config`